hydrolib/core/io/net/models.py

```diff
@@ -0,0 +1,916 @@
+
+        node_mask = self._mesh1d.get_node_mask(branchids)
+        if polygon is None:
+            polygon = self.meshkernel.mesh2d_get_mesh_boundaries_as_polygons()
+
+        self._link1d2d._link_from_1d_to_2d(node_mask, polygon=polygon)
+
+    def mesh2d_create_rectilinear_within_bounds(
+        self, extent: tuple, dx: float, dy: float
+    ) -> None:
+        self._mesh2d.create_rectilinear(extent=extent, dx=dx, dy=dy)
+
+    def mesh2d_create_triangular_within_polygon(self, polygon: mk.GeometryList) -> None:
+        raise NotImplementedError()
+
+    def mesh2d_clip_mesh(
+        self, polygon: mk.GeometryList, deletemeshoption: int = 1
+    ) -> None:
+        self._mesh2d.clip(polygon=polygon, deletemeshoption=deletemeshoption)
+
+    def mesh2d_refine_mesh(self, polygon: mk.GeometryList, level: int = 1) -> None:
+        self._mesh2d.refine(polygon=polygon, level=level)
+
+    def mesh1d_add_branch(
+        self,
+        branch: Branch,
+        name: str = None,
+        branch_order: int = -1,
+        long_name: str = None,
+    ) -> None:
+        self._mesh1d._add_branch(
+            branch=branch, name=name, branch_order=branch_order, long_name=long_name
+        )
+
+    def plot(
+        self,
+        ax: matplotlib.axes._subplots.AxesSubplot,
+        mesh1d_kwargs: dict = None,
+        mesh2d_kwargs: dict = None,
+        links1d2d_kwargs: dict = None,
+    ) -> None:
+
+        if mesh1d_kwargs is None:
+            mesh1d_kwargs = {"color": "C3", "lw": 1.0}
+        if mesh2d_kwargs is None:
+            mesh2d_kwargs = {"color": "C0", "lw": 0.5}
+        if links1d2d_kwargs is None:
+            links1d2d_kwargs = {"color": "k", "lw": 1.0}
+
+        # Mesh 1d
+        if not self._mesh1d.empty():
+            nodes1d = np.stack(
+                [self._mesh1d.mesh1d_node_x, self._mesh1d.mesh1d_node_y], axis=1
+            )
+            edge_nodes = self._mesh1d.mesh1d_edge_nodes
+            lc_mesh1d = LineCollection(nodes1d[edge_nodes], **mesh1d_kwargs)
+            ax.add_collection(lc_mesh1d)
+
+        # Mesh 2d
+        if not self._mesh2d.empty():
+            nodes2d = np.stack(
+                [self._mesh2d.mesh2d_node_x, self._mesh2d.mesh2d_node_y], axis=1
+            )
+            edge_nodes = self._mesh2d.mesh2d_edge_nodes
+            lc_mesh2d = LineCollection(nodes2d[edge_nodes], **mesh2d_kwargs)
+            ax.add_collection(lc_mesh2d)
+
+        # Links
+        if not self._link1d2d.empty():
+            faces2d = np.stack(
+                [self._mesh2d.mesh2d_face_x, self._mesh2d.mesh2d_face_y], axis=1
+            )
+            link_coords = np.stack(
+                [
+                    nodes1d[self._link1d2d.link1d2d[:, 0]],
+                    faces2d[self._link1d2d.link1d2d[:, 1]],
+                ],
+                axis=1,
+            )
+            lc_link1d2d = LineCollection(link_coords, **links1d2d_kwargs)
+            ax.add_collection(lc_link1d2d)
```

Comment:
I would move this to a separate utility which takes a `Network` as an argument, rather than have it as a property of the Network itself. This would allow us to drop the dependency on matplotlib for the core models, and instead localise all plotting related logic to a separate sub-model.